go/arrow/decimal128/decimal128.go

```diff
@@ -54,6 +59,36 @@ func FromI64(v int64) Num {
 	}
 }
 
+func fromBigIntPositive(v *big.Int) Num {
```

Comment:
Big int seems like it is arbitrary precision?  What happens if the value exceeds 16 bytes?

Reply:
added a check and panic if `v.BitLen()` > 128